### PR TITLE
Partially improve TypeScript compatibility tests

### PR DESCRIPTION
**Pull request by timostamm**
This partially gets us to the goals defined in #581. We do not remove the dependency on `@types/node` yet, but we make sure to use the corresponding version for every version of TypeScript we test with.

To speed up GitHub actions, the compatibility tests are moved to a separate job that runs in parallel.

### CODE REVIEW

File: packages/protobuf-example/package.json [hunk 1]
```diff
@@ -3,10 +3,9 @@
   "private": true,
   "license": "(Apache-2.0 AND BSD-3-Clause)",
   "scripts": {
-    "clean": "rm -rf ./dist/esm/* ./dist/types/*",
-    "build": "../../node_modules/typescript/bin/tsc --project tsconfig.json --module ES2015 --verbatimModuleSyntax --outDir ./dist/esm --declaration --declarationDir ./dist/types",
-    "add-person": "node dist/esm/add-person.js addressbook.bin",
-    "list-people": "node dist/esm/list-people.js addressbook.bin",
+    "build": "../../node_modules/typescript/bin/tsc --noEmit",
+    "add-person": "tsx src/add-person.ts addressbook.bin",
+    "list-people": "tsx src/list-people.ts addressbook.bin",
```
Comment (timostamm) [inline packages/protobuf-example/package.json:8]: A drive-by change. We have been using `tsx` in our examples everywhere but here.

**Review by smaye81 — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -25,4 +22,20 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-protobuf-es-ci-
       - name: make
-        run: make all checkdiff
```
Comment (miasahadat83-art) [inline .github/workflows/ci.yaml:28]: Nam ki

Comment (miasahadat83-art) [reply .github/workflows/ci.yaml:28]: কি করে সিল

File: packages/typescript-compat/v4.1.x/tsconfig.json [hunk 3]
```diff
@@ -1,12 +1,12 @@
 {
-  "include": ["../src/**/*"],
+  "include": ["../../protobuf-test/src/**/*"],
```
Comment (miasahadat83-art) [inline packages/typescript-compat/v4.1.x/tsconfig.json:2]: - [ ] ****

**Review by miasahadat83-art — COMMENTED** [review]
দেশের বাড়ি কই
